plugins/flashrom/fu-flashrom-device.c

```diff
@@ -8,11 +8,23 @@
 
 #include "fu-flashrom-device.h"
 
+#include <libflashrom.h>
+
 struct _FuFlashromDevice {
 	FuDevice		 parent_instance;
 };
 
-G_DEFINE_TYPE (FuFlashromDevice, fu_flashrom_device, FU_TYPE_DEVICE)
+typedef struct {
+	FuFlashromDevice		 parent_instance;
+	gsize				 flash_size;
+	struct flashrom_flashctx	*flashctx;
+	struct flashrom_layout		*layout;
+	struct flashrom_programmer	*flashprog;
+} FuFlashromDevicePrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE (FuFlashromDevice, fu_flashrom_device, FU_TYPE_DEVICE)
```

Comment:
Hangon, back a bit. You don't need to use G_DEFINE_TYPE_WITH_PRIVATE at all. The device is G_DECLARE_FINAL_TYPE, and so you can just stuff everything in the `G_DECLARE_FINAL_TYPE` above. Then you can use `self->flash_size` in the code rather than use `flash_size`

Reply:
Fixed